pkg/vulnsrc/suse-oval/suse-oval.go

```diff
@@ -0,0 +1,219 @@
+package suseoval
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"path/filepath"
+	"strings"
+
+	bolt "github.com/etcd-io/bbolt"
+
+	"github.com/aquasecurity/trivy-db/pkg/db"
+	"github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy-db/pkg/utils"
+	"github.com/aquasecurity/trivy-db/pkg/vulnsrc/vulnerability"
+	"golang.org/x/xerrors"
+)
+
+const (
+	platformOpenSUSEFormat   = "OpenSUSE Leap %s"
+	platformSUSELinuxFormat  = "SUSE Enterprise Linux %s"
+	ErrNoSuchFileOrDirectory = " no such file or directory"
+)
+
+var (
+	platformFormat = ""
+	suseDirs       = []string{
+		filepath.Join("oval", "suse", "opensuse.leap", "15.0"),
+		filepath.Join("oval", "suse", "opensuse.leap", "15.1"),
+		filepath.Join("oval", "suse", "opensuse.leap", "42.3"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise", "12"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise", "15"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise.desktop", "10"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise.desktop", "11"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise.desktop", "12"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise.desktop", "15"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise.server", "10"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise.server", "11"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise.server", "12"),
+		filepath.Join("oval", "suse", "suse.linux.enterprise.server", "15"),
+	}
+)
+
+type VulnSrc struct {
+	dbc db.Operations
+}
+
+func NewVulnSrc() VulnSrc {
+	return VulnSrc{
+		dbc: db.Config{},
+	}
+}
+
+func (vs VulnSrc) Update(dir string) error {
+	log.Println("Saving SUSE OVAL")
+
+	for _, suseDir := range suseDirs {
+		rootDir := filepath.Join(dir, "vuln-list", suseDir)
+		platformFormat = platformSUSELinuxFormat
```

Comment:
You can't aggregate SUSE Linux Enterprise Server and SUSE Linux Enterprise Server SP1 because they have different fixed versions.